src/utils/Eip1993Factories.ts

```diff
@@ -131,6 +133,27 @@ function fromEthersProvider(provider: EthersProvider): Provider {
   };
 }
 
+/**
+ * Creates a Provider instance from @zilliqa-js/core Provider
+ * @param provider - provider object
+ */
+function fromZilliqaProvider(provider: ZilliqaProvider): Provider {
```

Comment:
I don't think initializing Eip1993 provider from a Zilliqa one makes sense, it's unrelated to Ethereum.

Reply:
Converting Zilliqa provider to ethereum provider is the best way to integrate zilliqa provider. So even if you want to support zns initialization only with Zilliqa provider, you'd better convert it to eip provider under the hood to allow shared ethereum/zilliqa code like `FetchProvider` (a built-in provider used for both).

Reply:
I mean, this feels so wrong at least semantically, and I'm not quite convinced it's a good choice architecturally. Why can't we just implement `FetchProvider` for `ZilliqaProvider`? Imo, from the reader's perspective, the current suggested solution looks quite confusing and illogical.

Reply:
What use-case does it cover for users? And how exactly do we plan to use it for code sharing?

Reply:
```
new Resolution({
  zns: {provider: F. fromZilliqaProvider(zilliqa.provider)},
  cns: {provider: F.fromEthersProvider(ethers.provider)},
})
```

Reply:
Just to add one more point to our internal discussion, I'd be scared of a library that made me write something like this:
```
new Application({mysql: {connection: PostgresConnectionFactory.fromMySqlConnection(mysql.connection)}});
```